Inline Function in 'pio_rpi_pico.h' causes Linker Error when CONFIG_NO_OPTIMIZATIONS is enabled

### Describe the bug

The changes merged in PR #85167 introduced the inline function ```pio_rpi_pico_get_pio()```. When attempting to create an image for the Raspberry Pi Pico for debugging using the option ```CONFIG_NO_OPTIMIZATIONS=y```, this will cause a linker error.


### Regression

- [x] This is a regression.

### Steps to reproduce

Description for **Linux** build host.

1. Add a file to your Zephyr environment that contains the option for disabling optimizations: ```CONFIG_NO_OPTIMIZATIONS=y```. For example as ```~/debug.conf```
2. In the Zephyr tree run this command
```
west build -b rpi_pico --extra-conf ~/debug.conf samples/boards/raspberrypi/rpi_pico/uart_pio -- -DCMAKE_BUILD_TYPE=Debug
```
3. The compilation will fail during the linking stage.

### Relevant log output

```shell
[152/157] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/awolf/zephyrproject-fork/zephyr/build/zephyr/zephyr_pre0.map 
: && ccache /home/awolf/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -g -gdwarf-4 -O0 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map,/home/awolf/zephyrproject-fork/zephyr/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/libarch__arm__core.a  zephyr/arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a  zephyr/lib/libc/picolibc/liblib__libc__picolibc.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/lib/posix/c_lib_ext/liblib__posix__c_lib_ext.a  zephyr/drivers/misc/pio_rpi_pico/libdrivers__misc__pio_rpi_pico.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  zephyr/drivers/reset/libdrivers__reset.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/hal_rpi_pico/libmodules__hal_rpi_pico.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L/home/awolf/zephyrproject-fork/zephyr/build/zephyr  zephyr/arch/common/libisr_tables.a  -fuse-ld=bfd  -mcpu=cortex-m0plus  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -mtp=soft  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -Wl,--undefined=_sw_isr_table  -Wl,--undefined=_irq_vector_table  -specs=picolibc.specs  -DPICOLIBC_LONG_LONG_PRINTF_SCANF -L"/home/awolf/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/thumb/v6-m/nofp" -lc -lgcc && cd /home/awolf/zephyrproject-fork/zephyr/build/zephyr && /usr/bin/cmake -E true
/home/awolf/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/serial/libdrivers__serial.a(uart_rpi_pico_pio.c.obj): in function `pio_uart_poll_in':
/home/awolf/zephyrproject-fork/zephyr/drivers/serial/uart_rpi_pico_pio.c:117: undefined reference to `pio_rpi_pico_get_pio'
/home/awolf/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/serial/libdrivers__serial.a(uart_rpi_pico_pio.c.obj): in function `pio_uart_poll_out':
/home/awolf/zephyrproject-fork/zephyr/drivers/serial/uart_rpi_pico_pio.c:140: undefined reference to `pio_rpi_pico_get_pio'
/home/awolf/zephyr-sdk-0.17.4/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/serial/libdrivers__serial.a(uart_rpi_pico_pio.c.obj): in function `pio_uart_init':
/home/awolf/zephyrproject-fork/zephyr/drivers/serial/uart_rpi_pico_pio.c:153: undefined reference to `pio_rpi_pico_get_pio'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/awolf/zephyrproject-fork/zephyr/build
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- Host: Linux Ubuntu;
- Target: Raspberry Pi - All Pico and Pico2 variants;
- Latest code base after PR mentioned in description;


### Additional Context

This is a generally known issue with inline functions defined in 'C' headers.

A possible fix is to declare the function as ```extern``` within the 'C' implementation file ```spi_rpi_pico_pio.c```, in case the ```CONFIG_NO_OPTIMIZATIONS=y```:
```
#if defined(CONFIG_NO_OPTIMIZATIONS)
/* Force inline function to have a real reference for linking */
extern PIO pio_rpi_pico_get_pio(const struct device *dev);
#endif
```

--- a/include/zephyr/drivers/misc/pio_rpi_pico/pio_rpi_pico.h
+++ b/include/zephyr/drivers/misc/pio_rpi_pico/pio_rpi_pico.h
@@ -148,7 +148,7 @@
  * @param dev Pointer to device structure for rpi_pio device instance
  * @return PIO object
  */
-inline PIO pio_rpi_pico_get_pio(const struct device *dev)
+static inline PIO pio_rpi_pico_get_pio(const struct device *dev)
 {
 	return *(PIO *)(dev->config);
 }

